datahub-web-react/src/app/ingest/source/builder/RecipeForm/powerbi.ts

```diff
@@ -0,0 +1,172 @@
+import { get } from 'lodash';
+
+import { RecipeField, FieldType, setListValuesOnRecipe } from './common';
+
+export const POWERBI_CLIENT_ID: RecipeField = {
+    name: 'client_id',
+    label: 'Client ID',
+    tooltip: 'The Azure AD Client ID',
+    type: FieldType.TEXT,
+    fieldPath: 'source.config.client_id',
+    placeholder: 'client id',
+    required: true,
+    rules: null,
+};
+
+export const POWERBI_CLIENT_SECRET: RecipeField = {
+    name: 'client_secret',
+    label: 'Client Secret',
+    tooltip: 'The Azure AD Client Secret',
+    type: FieldType.SECRET,
+    fieldPath: 'source.config.client_secret',
+    placeholder: 'client secret',
+    required: true,
+    rules: null,
+};
+
+export const POWERBI_TENANT_ID: RecipeField = {
+    name: 'tenant_id',
+    label: 'Tenant ID',
+    tooltip: 'The Azure tenant id for your PowerBI instance.',
+    type: FieldType.TEXT,
+    fieldPath: 'source.config.tenant_id',
+    placeholder: 'a949d688-67c0-4bf1-a344-e939411c6c0a',
+    required: true,
+    rules: null,
+};
+
+const includeWorkspacesPath = 'source.config.extract_workspaces_to_containers';
+export const INCLUDE_WORKSPACES: RecipeField = {
+    name: 'extract_workspaces_to_containers',
+    label: 'Include Workspaces',
+    tooltip: 'Extract PowerBI Workspaces as Container nodes in DataHub.',
+    type: FieldType.BOOLEAN,
+    fieldPath: includeWorkspacesPath,
+    getValueFromRecipeOverride: (recipe: any) => {
+        const includeWorkspaces = get(recipe, includeWorkspacesPath);
+        if (includeWorkspaces !== undefined && includeWorkspaces !== null) {
+            return includeWorkspaces;
+        }
+        return true;
+    },
+    rules: null,
+};
+
+const adminsApisOnlyPath = 'source.config.admin_apis_only';
+export const ADMIN_APIS_ONLY: RecipeField = {
+    name: 'admin_apis_only',
+    label: 'Use Admin APIs Only',
+    tooltip:
+        'Only use the Admin APIs for extracting information from PowerBI. See the DataHub Ingestion Source Guide for more information.',
+    type: FieldType.BOOLEAN,
+    fieldPath: adminsApisOnlyPath,
+    getValueFromRecipeOverride: (recipe: any) => {
+        const adminApisOnly = get(recipe, adminsApisOnlyPath);
+        if (adminApisOnly !== undefined && adminApisOnly !== null) {
+            return adminApisOnly;
+        }
+        return false;
```

Comment:
nit: could make a common function like `getBooleanFieldWithDefault` and reuse with above in `INCLUDE_WORKSPACES`

Reply:
and the options below